File: gen_colors.py
import re

content = open('lib/services/prediction_repository.dart', 'r').read()

matches = re.finditer(r"id:\s*'([^']+)',\s*name:\s*'([^']+)',\s*shortName:\s*'([^']+)',\s*color:\s*Color\((0x[0-9a-fA-F]+)\)", content)

dart_file = '''import 'package:flutter/material.dart';

const Map<String, Color> kTeamColors = {
'''

seen = set()
for m in matches:
    team_id = m.group(1)
    name = m.group(2)
    short_name = m.group(3)
    color = m.group(4)
    if short_name not in seen:
        dart_file += f"  '{short_name}': Color({color}), // {name}\n"
        seen.add(short_name)

dart_file += '};\n'

with open('lib/data/team_colors.dart', 'w') as f:
    f.write(dart_file)

File: gen_colors.dart
import 'dart:io';

void main() {
  final content = File(
    'lib/services/prediction_repository.dart',
  ).readAsStringSync();
  final regex = RegExp(
    r"id:\s*'([^']+)',\s*name:\s*'([^']+)',\s*shortName:\s*'([^']+)',\s*color:\s*Color\((0x[0-9a-fA-F]+)\)",
  );
  final matches = regex.allMatches(content);

  final buffer = StringBuffer();
  buffer.writeln("import 'package:flutter/material.dart';");
  buffer.writeln();
  buffer.writeln("const Map<String, Color> kTeamColors = {");

  final seen = <String>{};
  for (final match in matches) {
    final shortName = match.group(3)!;
    if (!seen.contains(shortName)) {
      final name = match.group(2)!;
      final color = match.group(4)!;
      buffer.writeln("  '$shortName': Color($color), // $name");
      seen.add(shortName);
    }
  }

  buffer.writeln("};");
  File('lib/data/team_colors.dart').writeAsStringSync(buffer.toString());
}
